hazelcast-vm-cluster/azuredeploy.json

```diff
@@ -219,161 +200,103 @@
-        },
-        "osProfile": "[variables('osProfile')]",
-        "storageProfile": {
-          "imageReference": {
-            "publisher": "[variables('imagePublisher')]",
-            "offer": "[variables('imageOffer')]",
-            "sku": "[parameters('ubuntuOSVersion')]",
-            "version": "latest"
-          },
-          "osDisk": {
-            "name": "[concat(variables('vmName'), '0','_OSDisk')]",
-            "caching": "ReadWrite",
-            "createOption": "FromImage"
-          }
-        },
-        "networkProfile": {
-          "networkInterfaces": [
-            {
-              "id": "[resourceId('Microsoft.Network/networkInterfaces',concat(variables('nicName'), '0'))]"
-            }
+          "ipConfigurations": [
+              {
+                  "name": "[concat('ipconfig', copyIndex(1))]",
+                  "properties": {
+                      "privateIPAllocationMethod": "Dynamic",
+                      "publicIPAddress": {
+                          "id": "[resourceId('Microsoft.Network/publicIPAddresses', concat(variables('publicIPAddressName'), copyIndex(1)))]"
+                      },
+                      "subnet": {
+                          "id": "[variables('subnetRef')]"
+                      }
+                  }
+              }
           ]
-        }
-      }
+      },
+      "dependsOn": [
+          "[concat('Microsoft.Network/virtualNetworks/', variables('virtualNetworkName'))]",
+          "createPubIps"
+      ]
     },
     {
-      "apiVersion": "2017-03-30",
       "type": "Microsoft.Compute/virtualMachines",
       "name": "[concat(variables('vmName'), copyIndex(1))]",
+      "apiVersion": "2018-06-01",
       "location": "[parameters('location')]",
-      "dependsOn": [
-        "[concat('Microsoft.Network/networkInterfaces/', variables('nicName'), copyIndex(1))]",
-        "createNics"
-      ],
+      "copy": {
+          "name": "createVms",
+          "count": "[parameters('instanceCount')]"
+      },
       "tags": {
-        "[parameters('clusterName')]": "[parameters('clusterPort')]"
+          "[parameters('clusterTag')]": "[parameters('clusterPort')]"
       },
       "properties": {
-        "hardwareProfile": {
-          "vmSize": "[parameters('vmSize')]"
-        },
-        "osProfile": "[variables('osProfile')]",
-        "storageProfile": {
-          "imageReference": {
-            "publisher": "[variables('imagePublisher')]",
-            "offer": "[variables('imageOffer')]",
-            "sku": "[parameters('ubuntuOSVersion')]",
-            "version": "latest"
+          "hardwareProfile": {
+              "vmSize": "[parameters('vmSize')]"
+          },
+          "osProfile": {
+            "computerName": "[variables('vmName')]",
+            "adminUsername": "[parameters('adminUsername')]",
+            "adminPassword": "[if(equals(parameters('authenticationType'), 'sshPublicKey'), json('null'), parameters('adminPasswordOrKey'))]",
```

Comment:
just use:

"adminPassword": "[parameters('adminPasswordOrKey')]",

since password authn is disabled in the key config

Reply:
Hi @bmoore-msft ,
I checked [101-vm-simple-linux](https://github.com/Azure/azure-quickstart-templates/tree/master/101-vm-simple-linux) which has similar configuration but it does not work with sshPublicKey. Apparently, we fixed an existing problem which sshPublicKeys are not allowed during validation

Here is the PR for the fix.
https://github.com/Azure/azure-quickstart-templates/pull/5804

Reply:
Not sure what you're seeing but the current syntax works find with either config, so no need to change that behavior.

Reply:
That is what I see when I use 101-vm-simple-linux and paste my public key.
https://github.com/Azure/azure-quickstart-templates/pull/5804#issuecomment-464684925

Reply:
@bmoore-msft I also tried [101-vm-simple-linux](https://github.com/Azure/azure-quickstart-templates/tree/master/101-vm-simple-linux) and I have encountered same issue at https://github.com/Azure/azure-quickstart-templates/pull/5804#issuecomment-464684925

Do you have any fix/solution for this password error?

Reply:
Hi @bmoore-msft ,
Is this the comment that is blocking the merge of this PR?
Unfortunately, last time I tried to use ssh key on `101-vm-simple-linux`, I was not able to do it. Apparently, @hasancelik  is also in the same situation. 

You can see the screenshots on my PR comment.
https://github.com/Azure/azure-quickstart-templates/pull/5804#issuecomment-464684925